src/Storefront/Controller/CartLineItemController.php

```diff
@@ -276,6 +276,9 @@ private function traceErrors(Cart $cart): bool
      */
     private function getLineItemArray(RequestDataBag $lineItemData): array
     {
+        if ($lineItemData->has('payload')) {
+            $lineItemData->set('payload', json_decode($lineItemData->getString('payload'), true));
```

Comment:
Can you provide a small unit test for this?

Reply:
And done ;)